File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--validate-data-when-a-new-row-is-added-to-a-listobject-control_2.cs
            employeeTable = new System.Data.DataTable("Employees");

            System.Data.DataColumn column = employeeTable.Columns.Add
                ("Id", typeof(int));
            column.AllowDBNull = false;

            employeeTable.Columns.Add("FirstName", typeof(string));
            employeeTable.Columns.Add("LastName", typeof(string));
            employeeTable.Columns.Add("Age", typeof(int));

            employeeTable.Rows.Add(id, "Nancy", "Anderson", "56");
            employeeTable.Rows.Add(id, "Robert", "Brown", "44");
            id++;

            list1.SetDataBinding(employeeTable, "", "FirstName", "LastName", "Age");

            list1.BeforeAddDataBoundRow +=new Microsoft.Office.Tools.Excel.
                BeforeAddDataBoundRowEventHandler(list1_BeforeAddDataBoundRow);

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--validate-data-when-a-new-row-is-added-to-a-listobject-control_2.vb
        employeeTable = New System.Data.DataTable("Employees")

        Dim column As System.Data.DataColumn = _
            employeeTable.Columns.Add("Id", GetType(Int32))
        column.AllowDBNull = False

        employeeTable.Columns.Add("FirstName", GetType(String))
        employeeTable.Columns.Add("LastName", GetType(String))
        employeeTable.Columns.Add("Age", GetType(Int32))

        employeeTable.Rows.Add(id, "Nancy", "Anderson", 56)
        employeeTable.Rows.Add(id, "Robert", "Brown", 44)
        id += 1

        list1.SetDataBinding(employeeTable, "", "FirstName", "LastName", "Age")